src/main/c/ibs.h

```diff
@@ -0,0 +1,62 @@
+#ifndef HAIL_OVERRIDE_ARCH
+#if __AVX2__
+#define SIMDPP_ARCH_X86_AVX2
+#elif __AVX__
+#define SIMDPP_ARCH_X86_AVX
+#elif __SSE4_1__
+#define SIMDPP_ARCH_X86_SSE4_1
+#elif __SSSE3__
+#define SIMDPP_ARCH_X86_SSSE3
+#elif __SSE3__
+#define SIMDPP_ARCH_X86_SSE3
+#elif __SSE2__
+#define SIMDPP_ARCH_X86_SSE2
+#endif
+#endif // HAIL_OVERRIDE_ARCH
+
+#include <simdpp/simd.h>
+#include <inttypes.h>
+
+using namespace simdpp;
+
+#ifndef HAIL_OVERRIDE_WIDTH
+#define HAIL_VECTOR_WIDTH SIMDPP_FAST_INT64_SIZE
+#else
+#define HAIL_VECTOR_WIDTH HAIL_OVERRIDE_WIDTH
+#endif // HAIL_OVERRIDE_WIDTH
+
+using hailvec = uint64<HAIL_VECTOR_WIDTH>;
+
+#ifndef CACHE_SIZE_PER_MATRIX_IN_KB
+#define CACHE_SIZE_PER_MATRIX_IN_KB 4
+#endif
+
+#ifndef CACHE_SIZE_IN_MATRIX_ROWS
+#define CACHE_SIZE_IN_MATRIX_ROWS 4 * CACHE_SIZE_PER_MATRIX_IN_KB
```

Comment:
I cleaned up the code a fair bit, but then made this bit more complicated. I think, however, that it's more straightforward now? Let me know what you think.